### PR TITLE
chore: adds color explorer package

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description

Adds color explorer source to fast ecosystem to aid in color testing

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-color-explorer/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Introduction 
+Application for UI testing and development.
+
+# Getting Started
+Clone the repository, and then run `npm i` in the root directory.
```
Comment (janechu) [inline packages/fast-color-explorer/README.md]: do you want to say to run `lerna bootstrap` then cd into the packages/fast-color-explorer?

File: packages/fast-color-explorer/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Introduction 
+Application for UI testing and development.
+
+# Getting Started
```
Comment (janechu) [inline packages/fast-color-explorer/README.md]: if you use a single "#" these will all show as the equivalent of h1, so this, and the ones below, should be ## or greater.

File: packages/fast-color-explorer/app/app.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    };
+
+    private renderColorBlock = (props: any): JSX.Element => {
+        const color: string = palette(PaletteType.neutral)(this.props.designSystem)[
+            props.index
+        ];
+
+        return (
+            <div style={props.style} key={color}>
+                <ColorBlocks {...{ backgroundColor: color, index: props.index } as any} />
```
Comment (janechu) [inline packages/fast-color-explorer/app/app.tsx]: shouldn't this just be `<ColorBlocks backgroundColor={color} index={props.index} />`?

File: packages/fast-color-explorer/app/color-blocks.tsx [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+import { get, isEqual, uniqueId } from "lodash-es";
+import { AppState, ComponentTypes } from "./state";
+import { connect } from "react-redux";
+import { ButtonClassNameContract } from "@microsoft/fast-components-react-base";
+import {
+    FillSwatchFamily,
+    SwatchFamily,
+} from "@microsoft/fast-components-styles-msft/dist/utilities/color/common";
+
+const styles: any = {
```
Comment (janechu) [inline packages/fast-color-explorer/app/color-blocks.tsx]: can this be something other than any?

File: packages/fast-color-explorer/app/color-blocks.tsx [hunk 5]
```diff
@@ -0,0 +1,581 @@
...
+                designSystem: Object.assign({}, props.designSystem, {
+                    backgroundColor: props.backgroundColor,
+                }),
+            };
+        }
+
+        return null;
+    }
+
+    private titleStyleOverrides: any = {
```
Comment (janechu) [inline packages/fast-color-explorer/app/color-blocks.tsx]: can this be something other than any?

File: packages/fast-color-explorer/app/color-blocks.tsx [hunk 6]
```diff
@@ -0,0 +1,581 @@
...
+            >
+                <div className={this.props.managedClasses.colorBlocks_title}>
+                    {this.props.index}
+                </div>
+                <div className={this.props.managedClasses.colorBlocks_backgroundColor}>
+                    <Caption>
+                        {this.state.designSystem.backgroundColor.toUpperCase()}
+                    </Caption>
+                </div>
+
```
Comment (janechu) [inline packages/fast-color-explorer/app/color-blocks.tsx]: nit: extra line

File: packages/fast-color-explorer/app/color-blocks.tsx [hunk 7]
```diff
@@ -0,0 +1,581 @@
...
+/* tslint:disable-next-line */
+const ColorBlocks = manageJss(styles)(ColorBlocksBase);
+type ColorBlocks = InstanceType<typeof ColorBlocks>;
+
+function mapStateToProps(state: AppState): Partial<ColorBlocksProps> {
+    return {
+        component: state.componentType,
+        designSystem: state.designSystem,
+    };
```
Comment (janechu) [inline packages/fast-color-explorer/app/color-blocks.tsx]: could just do `return { ...state };` if you matched the props to state for component/componentType

File: packages/fast-color-explorer/app/control-pane.tsx [hunk 8]
```diff
@@ -0,0 +1,381 @@
...
+
+        this.state = {
+            accentColorBase:
+                defaultAccentBase instanceof ColorRGBA64
+                    ? defaultAccentBase.toStringHexRGB()
+                    : AccentColors.blue,
+            neutralColorBase:
+                defaultNeutralBase instanceof ColorRGBA64
+                    ? defaultNeutralBase.toStringHexRGB()
+                    : "#808080",
```
Comment (janechu) [inline packages/fast-color-explorer/app/control-pane.tsx]: seems like this value is captured in your neutralShortcuts, should you use that instead?

File: packages/fast-color-explorer/app/control-pane.tsx [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+                )}
+            >
+                {items}
+            </div>
+        );
+    }
+
+    private handleNeutralBaseColorChange = (
+        c: any,
+        e: React.ChangeEvent<HTMLInputElement>
```
Comment (janechu) [inline packages/fast-color-explorer/app/control-pane.tsx]: can we use longer names for these arguments?

File: packages/fast-color-explorer/app/control-pane.tsx [hunk 10]
```diff
@@ -0,0 +1,381 @@
...
+            accentColorBase: c.hex,
+        });
+    };
+}
+
+function mapStateToProps(state: AppState): Partial<ControlPaneProps> {
+    return {
+        componentType: state.componentType,
+        designSystem: state.designSystem,
+    };
```
Comment (janechu) [inline packages/fast-color-explorer/app/control-pane.tsx]: could do same thing i suggested above, `return { ...state };`

File: packages/fast-color-explorer/app/design-system.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+} from "@microsoft/fast-components-styles-msft";
+import { ColorRGBA64, parseColorHexRGB } from "@microsoft/fast-colors";
+import { createColorPalette } from "./palette";
+
+export const accentColors: string[] = [
+    "#0078d4",
+    "#107c10",
+    "#5c2d91",
+    "#d83b01",
+    "#f2c812",
```
Comment (janechu) [inline packages/fast-color-explorer/app/design-system.ts]: nit: should we be consistent with uppercase alphabet for HEX?

File: packages/fast-color-explorer/app/swatch.tsx [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    showBackgroundContrast?: boolean;
+
+    /**
+     * Show contrast between color and background color
+     */
+    showTextContrast?: boolean;
+}
+
+const styles: any = {
```
Comment (janechu) [inline packages/fast-color-explorer/app/swatch.tsx]: really any?

File: packages/fast-color-explorer/app/app.tsx (hunk 3 above)
Comment (nicholasrice) [reply packages/fast-color-explorer/app/app.tsx]: yea but react-redux is doing somthing funny with the typings and that errors.

File: packages/fast-color-explorer/app/control-pane.tsx [hunk 13]
```diff
@@ -0,0 +1,349 @@
...
+        },
+        controlPane: {
+            position: "relative",
+            zIndex: "1",
+            padding: "12px",
+            boxSizing: "border-box",
+            color: neutralForegroundRest,
+            height: "100%",
+            maxWidth: "300px",
+            boxShadow: (config: ColorsDesignSystem): string => {
```
Comment (bheston) [inline packages/fast-color-explorer/app/control-pane.tsx]: This could be applyElevation now.

File: packages/fast-color-explorer/app/swatch.tsx [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+        height: "36px",
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        margin: "4px 0",
+        transition: "background-color .05s ease-in",
+        width: "180px",
+        fontSize: "12px",
+        borderRadius: "18px",
+        boxShadow: (elevation(ElevationMultiplier.e2, "#000")(undefined as any) as any)
```
Comment (bheston) [inline packages/fast-color-explorer/app/swatch.tsx]: Can be applyElevation.

File: packages/fast-color-explorer/app/control-pane.tsx (hunk 13 above)
Comment (nicholasrice) [reply packages/fast-color-explorer/app/control-pane.tsx]: I'll clean up in another PR - just trying to migrate.

File: packages/fast-color-explorer/app/swatch.tsx (hunk 14 above)
Comment (nicholasrice) [reply packages/fast-color-explorer/app/swatch.tsx]: I'll clean up in another PR - just trying to migrate.

**Pull request merged**